# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546405944 28800
#      Tue Jan 01 21:12:24 2019 -0800
# Branch stable
# Node ID bb3ebaa9460e4abfac4ed5e8a12290500232abd7
# Parent  409120c6bf4349ef58e9c8a66e55b1e8a95c481c
psi.cc: Fix static analyzer detected issues (bug #55347).

* psi.cc (Fpsi): Call args(2).xidx_type_value()
rather than incorrect version without prepended 'x'.

diff --git a/libinterp/corefcn/psi.cc b/libinterp/corefcn/psi.cc
--- a/libinterp/corefcn/psi.cc
+++ b/libinterp/corefcn/psi.cc
@@ -68,17 +68,17 @@ can have any value real or complex value
 @end deftypefn */)
 {
   int nargin = args.length ();
 
   if (nargin < 1 || nargin > 2)
     print_usage ();
 
   const octave_value oct_z = (nargin == 1) ? args(0) : args(1);
-  const octave_idx_type k = (nargin == 1) ? 0 : args(0).idx_type_value ("psi: K must be an integer");
+  const octave_idx_type k = (nargin == 1) ? 0 : args(0).xidx_type_value ("psi: K must be an integer");
   if (k < 0)
     error ("psi: K must be non-negative");
 
   octave_value retval;
 
   if (k == 0)
     {
 #define FLOAT_BRANCH(T, A, M, E)                                \
